caas/kubernetes/provider/customresourcedefinitions.go

```diff
@@ -19,57 +28,73 @@ func (k *kubernetesClient) getCRDLabels(appName string) map[string]string {
 	}
 }
 
+func (k *kubernetesClient) getCRLabels(appName string) map[string]string {
+	return map[string]string{
+		labelApplication: appName,
+	}
+}
+
 // ensureCustomResourceDefinitions creates or updates a custom resource definition resource.
-func (k *kubernetesClient) ensureCustomResourceDefinitions(appName string, crds map[string]apiextensionsv1beta1.CustomResourceDefinitionSpec) (cleanUps []func(), _ error) {
-	for name, crd := range crds {
-		crd, err := k.ensureCustomResourceDefinition(name, k.getCRDLabels(appName), crd)
+func (k *kubernetesClient) ensureCustomResourceDefinitions(
+	appName string, crdSpecs map[string]apiextensionsv1beta1.CustomResourceDefinitionSpec,
+) (cleanUps []func(), _ error) {
+	for name, spec := range crdSpecs {
+		crd := &apiextensionsv1beta1.CustomResourceDefinition{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      name,
+				Namespace: k.namespace,
+				Labels:    k.getCRDLabels(appName),
```

Comment:
need annotations here too - see other places where we get them from the params.ResourceTags and then pass them in to the various ensure() functions